File: languages/python/bloock/entity/identity/boolean_attribute_descriptor.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto
from bloock.entity.identity.attribute_descriptor import AttributeDescriptor


class BooleanAttributeDescriptor(AttributeDescriptor):
    @staticmethod
    def from_proto(a: proto.BooleanAttributeDefinition) -> BooleanAttributeDescriptor:
        return BooleanAttributeDescriptor(
            display_name=a.display_name,
            technical_name=a.id,
            description=a.description,
        )

    def to_proto(self) -> proto.BooleanAttributeDefinition:
        return proto.BooleanAttributeDefinition(
            display_name=self.display_name,
            id=self.technical_name,
            description=self.description,
        )

File: languages/php/Bloock/Entity/Identity/BooleanAttributeDescriptor.php
<?php

namespace Bloock\Entity\Identity;

use Bloock\BooleanAttributeDefinition;

/**
 * Represents a descriptor for a boolean attribute.
 */
class BooleanAttributeDescriptor extends AttributeDescriptor
{
    public static function fromProto(BooleanAttributeDefinition $res): BooleanAttributeDescriptor
    {
        return new BooleanAttributeDescriptor($res->getDisplayName(), $res->getId(), $res->getDescription(), $res->getRequired());
    }

    public function toProto(): BooleanAttributeDefinition
    {
        $p = new BooleanAttributeDefinition();
        $p->setDisplayName($this->displayName);
        $p->setId($this->technicalName);
        $p->setDescription($this->description);
        $p->setRequired($this->required);
        return $p;
    }
}
